### ISSUE TITLE
MLVApp v1.0 Dual ISO preview broken (since v0.17) & export not processed at all?

### ISSUE THREAD

**Issue by gac123** [ilia3101/MLV-App#107]
PART 1: Broken Preview
Here is the file (shot with 5d mk III,720p, 25fps, mlv 14bit lossless, iso 100/3200) https://mega.nz/#F!V1IiEKqC!K-YCjrQ7LOKwdXX5fv17xQ (M03-1222.MLV)

So, the processed preview works correctly when the Dual ISO under RAW Correction is first set to On.
When scrubbing through the footage, few frames might be ok but at a certain point the unprocessed dual iso shows. At that point switching Dual ISO mode off and back on does nothing to get the correct preview back.

Then weird streaks start to appear. For example, on frame 230. (see image weirdstreaks.png)

To get a working preview, the file has to be reimported into the app. In v0.15 it works much better.

PART2: 
Say you start from scratch, import file, set Dual ISO option to On, check export settings ( cdng fast pass, export audio), export selected clip, the resulting dng's are unprocessed, (see image M03-1222_000000.dng) eventhough the preview is showing a correctly processed Dual ISO preview and the Enable RAW Correction is on.

I have tried this with the following export settings, all of them failing to contain dual iso processed frames.
- cdng fast pass
- cdng lossless
- cdng uncompressed
- prores4444 (standard options)

so is the Dual ISO feature broken or am I missing something? I am not finding any complaints on the magic lantern forum or elsewhere with my google-fu

**Comment by bouncyball-git** [comment]
Which OS?

Fast pass will always bypass any processing so it is expected. Will check other cases.

**Comment by gac123** [comment]
for me personally, this test I made was to make sure the dual iso is robust enough to improve the recordings i make of concert lightshow

now i'm shooting with magic lantern raw 14bit lossless on a 128gb card 720p (fills the card right up for that show), but I would love the increased dynamic range dual iso could provide without the ghosting effect hdr video introduces, even though i have not successfully gotten any workflow that results in any hdr footage, seems this technique is not really pursuit anymore (since 2013-2015), so i'm trying this dual iso way

**Comment by bouncyball-git** [comment]
Well... I understand your point. Then you have to experiment with dual iso. I have this bug on my fix list, just waiting some stuff from g3gg0 to be merged to the 4k branch of ML.

**Comment by gac123** [comment]
quick question though.. why does it break after frame 288? I mean, why does it have to detect anything? aren't we just saying, ok this is dual iso footage, now process the odd/even lines to expand the dynamic range, what does it have to detect? ELI5 please :)
oh, and great work with v1.0 it seems to import my 1,5hours footage much faster now!

**Comment by bouncyball-git** [comment]
Both preview (from dmilligan and full 20bit dual iso from a1ex) are detecting presence of different iso lines in the raw buffer. 

Also there was no DISO (dual iso MLV info block) in MLVs until recent changes from g3gg0 (to be fully merged yet), so mlvapp has to detect dual iso and process accordingly (to not process non dual iso frame with dual iso method b/c there may be some issues, even crash)

**Comment by gac123** [comment]
Not sure if I understand because still I'm wondering, why the need to detect/prevent dual iso method on non-dual iso footage, I understand making things monkey-proof, but this is pretty advanced stuff, or even get a particular look using non-diso footage.

on the preview: if it has to detect diso then doesn't that make the dual iso button redundant?

import footage>anything suggesting this is diso footage?>if yes>assume all frames are diso>else>assume non-diso

I'm probably oversimplifying this but for the little programming i do (local automatisation with python, and the typical webdev languages), this is what i'm wondering; making an educated guess/assumption once prevents taxing the system for this constant detection

please tell me to shut up if i'm wasting your time :)

**Comment by bouncyball-git** [comment]
> mport footage>anything suggesting this is diso footage?>if yes>assume all frames are diso>else>assume non-diso

That is the exact problem we can not detect dual iso right after importing and having only metadata because DISO info was not implemented until now (all older MLVs are w/o DISO block) and we have to analyse frame data itself to be sure and set dual iso flag in mlv app.

The detection will happen in lib anyway because otherwise a1ex's algo not gonna work. If you allow to process non real dual iso data with dual iso method in the best case you are wasting CPU and time (a lot!) and in the worst case you get crash.

**Comment by bouncyball-git** [comment]
I fixed this issue [here](https://github.com/ilia3101/MLV-App/commit/892b6cfbfe0c04643dea0b4ce7058d7bbce601d0).

The processing is off if exposure matching is failed. Now dual iso processing forced if exposures are not matched too. This corrects behavior explained above and MLV plays fine to the end.
